desktop/core/src/desktop/api2_tests.py

```diff
@@ -699,6 +699,29 @@ def test_create(self):
         json.loads(Document2.objects.get(type='gist', uuid=gist2['uuid']).data)['statement_raw']
     )
 
+    # Create multiple gist directories and then make a new gist
+    home_dir = Directory.objects.get_home_directory(self.user)
+    gist_dir2 = Directory.objects.create(name='Gist', owner=self.user, parent_directory=home_dir)
+
+    assert_equal(2, Directory.objects.filter(name='Gist', type='directory').count())
+
+    response3 = self._create_gist(
+        statement='SELECT 3',
+        doc_type='hive-query',
+        name='test_gist_create3',
+    )
+    gist3 = json.loads(response3.content)
+
+    # Gist directories merged into one
+    assert_equal(1, Directory.objects.filter(name='Gist', type='directory').count())
+
+    assert_true(Document2.objects.filter(type='gist', name='test_gist_create3'))
```

Comment:
`.exists()` or `get()` is more explicit if we want to test for only 1 unique doc

Reply:
still

Reply:
other comments as well

Reply:
Done